### ISSUE TITLE
Editor: fix fatal javascript errors

### ISSUE THREAD

**Issue by JoeGermuska** [NUKnightLab/StoryMapJS#273]
Users are reporting errors when trying to use the StoryMap editor. The most obvious error message in an editor console was `Uncaught TypeError: this.LatLngBounds is not a function` at [`EditorMap.js:41`](https://github.com/NUKnightLab/StoryMapJS/blob/fe7c935/static/js/EditorMap.js#L41)

There are also errors from the Google Maps API, and I'm wondering if the issue traces back to changes in that API. It doesn't seem like google would just change a running API, but there are errors when that loads:
![image](https://cloud.githubusercontent.com/assets/46313/13066488/0624b7ae-d42c-11e5-9a0f-b0aee4961154.png)

**Comment by JoeGermuska** [comment]
The Google Maps API is a strange legacy since most of StoryMap uses Leaflet; we chose to Google Maps in the editor in order to offer a place search. A bit of experimenting in the [`editor-fix`])https://github.com/NUKnightLab/StoryMapJS/tree/editor-fix) branch to remove loading Google Maps did get past the issue.

**Option 1: Remove Google Maps from the editor**
This would involve continuing down the path I experimented with tonight. Some likely necessary steps:

- make the [`LeafletEditorMap.setMapType()`](https://github.com/NUKnightLab/StoryMapJS/blob/editor-fix/static/js/EditorMap.js#L513-L521) function actually pay attention to the selected map type and create the right kind of tile layer -- now it just uses OSM
- make the place search box which appears for non-Gigapixel maps work, either by loading a newer version of the Google API JS or by using Mapbox's reverse geocoder (which may not work for non-addresses yet?)
- Look for other ways that `LeafletEditorMap` has embedded assumptions that it's being used for Zoomify that are just accidents of history, similar to the mixed method invocation signatures for `setMapType` that [I had to fix](https://github.com/NUKnightLab/StoryMapJS/commit/bca252d8eb6fdba4fdad83bc02b668add236f507), or the assumption that the map's `tilelayer` would have a `getCenterZoom()` method.

**Option 2: Update to use newer Google JS API**
I looked at this for a minute. Following the path laid out in the docs, including loading the JS with `async defer` is tricky because the existing code isn't well factored for having an init method to call when the Google API JS is loaded, and loading it synchronously ... well, maybe it would work out but I got annoyed and didn't pursue it.
